### DISCUSSION TITLE
LevensteinDistance function throws an exception on empty string(s)

### DISCUSSION THREAD

**Discussion by lmores** [duckdb/duckdb#5061] [Q&A]
The current implementation of the `LevenshteinDistance(s1, s2)` function throws an exception if either `s1` or `s2` is an empty string, see [the code](https://github.com/duckdb/duckdb/blob/a0ce35f0b5f77d2a5340bdab1fa4b42ec63cd722/src/function/scalar/string/levenshtein.cpp#L19).

I do not think this is the correct behaviour, as the Levenshtein distance is defined also for empty strings, as explained in the [same link](https://en.wikipedia.org/wiki/Levenshtein_distance) that appears in the description of the function.

Would you consider a patch that, when a string is empty, immediately returns the length of the other string?

**Comment by Mytherin** [comment]
> Would you consider a patch that, when a string is empty, immediately returns the length of the other string?
 
Sure, happy to review a PR
